tests/migrations/test_autodetector.py

```diff
@@ -2044,19 +2032,19 @@ def test_circular_dependency_swappable_self(self):
         self.assertOperationTypes(changes, 'a', 0, ["CreateModel"])
         self.assertMigrationDependencies(changes, 'a', 0, [])
 
-    def test_add_blank_textfield_and_charfield(self):
+    @mock.patch('django.db.migrations.questioner.MigrationQuestioner.ask_not_null_addition',
+                side_effect=Exception("Should not have prompted for not null addition"))
+    def test_add_blank_textfield_and_charfield(self, mocked_ask_method):
         """
         #23405 - Adding a NOT NULL and blank `CharField` or `TextField`
         without default should not prompt for a default.
         """
-        class CustomQuestioner(MigrationQuestioner):
-            def ask_not_null_addition(self, field_name, model_name):
-                raise Exception("Should not have prompted for not null addition")
-
         before = self.make_project_state([self.author_empty])
         after = self.make_project_state([self.author_with_biography_blank])
-        autodetector = MigrationAutodetector(before, after, CustomQuestioner())
+        autodetector = MigrationAutodetector(before, after)
         changes = autodetector._detect_changes()
+        self.assertEqual(mocked_ask_method.call_count, 0)
```

Comment:
I don't think this is useful to test 0 call count, because if questioner will be called, it will produce side effect (for this case).
It makes sense for cases where questionnaire is actually called